source/DistanceAndDirection/ProAppDistanceAndDirectionModule/ProAppDistanceAndDirectionModule/ViewModels/ProCircleViewModel.cs

```diff
@@ -539,14 +540,44 @@ public override string DistanceString
             {
                 if (CircleType == CircleFromTypes.Diameter)
                 {
-                    return (Distance * 2.0).ToString("G");
+                    if (EndsWithDecimal)
+                        return (Distance * 2.0).ToString("G") + ".";
+                    else if(EndsWithComma)
+                        return (Distance * 2.0).ToString("G") + ",";
+                    else
+                        return (Distance * 2.0).ToString("G");
                 }
-
-                return base.DistanceString;
+                if (EndsWithDecimal)
+                    return base.DistanceString + ".";
+                else if(EndsWithComma)
+                    return base.DistanceString + ",";
+                else
+                    return base.DistanceString;
             }
             set
             {
-                // lets avoid an infinite loop here
+                //Handle for decimals
+                if(value.EndsWith("."))
+                {
+                    EndsWithDecimal = true;
+                    return;
+                }
+                else
+                {
+                    EndsWithDecimal = false;
+                }
+
+                if (value.EndsWith(","))
```

Comment:
It might be better to use (and set on startup) CurrentCulture.NumberFormat.NumberDecimalSeparator